tests/js/server/aql/aql-optimizer-rule-optimize-cluster-single-document-operations.js

```diff
@@ -587,8 +587,104 @@ function optimizerClusterSingleDocumentTestSuite () {
         let result = AQL_EXPLAIN(query, { "@cn1" : cn1 });
         assertNotEqual(-1, result.plan.rules.indexOf(ruleName), query);
       });
-    }
+    },
 
+    testStatsWhenApplyingRuleOneAtATime: function () {
+      const collName = "testCollectionStats";
+      const doc = {value1: 1, value2: "a"};
+      const values = [1, 3];
+      values.forEach(numShards => {
+        try {
+          db._create(collName, {numberOfShards: numShards});
+          for (let i = 0; i < 10; ++i) {
```

Comment:
What is the benefit of running all following operations 10 times? Shouldn't the results of all 10 iterations be exactly identical? I think the for loop is unnecessary here, unless I am missing something.